File: 프로그래머스/1/131128. 숫자 짝꿍/숫자 짝꿍.py
def solution(X, Y):
    num_lst = '9876543210'
    result = ''
    
    for i in num_lst:
        min_i_count = min(X.count(i), Y.count(i))
        for j in range(min_i_count):
            result += i
            
    if len(result) == 0: return '-1'
    if result[0] == '0': return '0'

    return result

File: 프로그래머스/1/131128. 숫자 짝꿍/숫자 짝꿍.js
const solution = (X, Y) => {
    const count_X = Array(10).fill(0);
    const count_Y = Array(10).fill(0);
    
    for (const x of X) { count_X[x] ++; }
    for (const y of Y) { count_Y[y] ++; }
    
    let result = ''
    for (let i=9; i>-1; i--) {
        const cur_count = Math.min(count_X[i], count_Y[i]);
        result += String(i).repeat(cur_count);
    }
    if (result[0] === '0') { return '0'; }
    return !result ? '-1' : result;
}